### PR TITLE
Add rollup replace plugin for env vars subst.

### PR DESCRIPTION
**Pull request by ssipos90**
Hello,

I've added the ability to provide some of vars as env variables at rollup time. We are using this on our private OSM instance in a dockerized environment.

Also, I refactored the rollup config to prevent duplicate code from accumulating.

### CODE REVIEW

File: modules/config.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+/* eslint-disable no-undef */
```
Comment (ssipos90) [inline modules/config.js:1]: This file is necessary to avoid the eslint rule, otherwise is redundant.

File: modules/services/osm.js [hunk 2]
```diff
@@ -12,14 +12,15 @@ import { geoExtent, geoRawMercator, geoVecAdd, geoZoomToScale } from '../geo';
 import { osmEntity, osmNode, osmNote, osmRelation, osmWay } from '../osm';
 import { utilArrayChunk, utilArrayGroupBy, utilArrayUniq, utilRebind, utilTiler, utilQsString } from '../util';
 
+import { osm_urlroot, oauth_consumer_key, oauth_secret } from '../config';
 
 var tiler = utilTiler();
 var dispatch = d3_dispatch('apiStatusChange', 'authLoading', 'authDone', 'change', 'loading', 'loaded', 'loadedNotes');
-var urlroot = 'https://www.openstreetmap.org';
+var urlroot = osm_urlroot;
```
Comment (ssipos90) [inline modules/services/osm.js:19]: This var is duped because there's a setting down somewhere that can override this and importing creates const.

**Comment by ssipos90** [comment]
Not worth it.

**Comment by quincylvania** [comment]
> Not worth it.

@ssipos90 I haven't had a chance to look at this yet, but I'd be happy to work through it with you!

**Comment by ssipos90** [comment]
I closed it because it's not the best approach regarding the auth mechanism, however it can be used for other build time vars.

`osmAuth` should be configurable at runtime (overriding the `urlroot` and `oauth_*` stuff when creating the context by calling the `switch` function or smtn) and this is something that I plan working on in the near future.

**Comment by bhousel** [comment]
I would prefer to not refactor the config files.  I know there is duplicate code there to support different build targets, but I think it's valuable to be able to open a config file and see everything that it does without needing to look elsewhere.

**Comment by ssipos90** [comment]
I've created another PR (#7720) that doesn't refactor Rollup config files, as requested.

**Pull request closed without merging**